File: static/java/FormulaField.java
// ==========================================
// MyAirtable static runtime.
// ==========================================

package myairtable;

/**
 * Base interface for all formula field types. Provides field reference and empty/not-empty checks.
 */
public interface FormulaField {

  String getFieldId();

  /** Airtable field reference: {@code {fieldId}}. */
  default String field() {
    return "{" + getFieldId() + "}";
  }

  /** Field is blank. */
  default String empty() {
    return field() + "=BLANK()";
  }

  /** Field is not blank. */
  default String notEmpty() {
    return field() + "!=BLANK()";
  }
}

File: static/csharp/FormulaField.cs
namespace MyAirtable;

/// <summary>Base class for all formula field builders: field reference + empty/not-empty checks.</summary>
public abstract class FormulaField
{
    protected FormulaField(string fieldId) => FieldId = fieldId;

    public string FieldId { get; }

    /// <summary>Airtable field reference: <c>{fieldId}</c>.</summary>
    public string Field() => "{" + FieldId + "}";

    /// <summary>Field is blank.</summary>
    public virtual string Empty() => Field() + "=BLANK()";

    /// <summary>Field is not blank.</summary>
    public virtual string NotEmpty() => Field() + "!=BLANK()";
}
